src/test/groovy/nebula/plugin/dependencyverifier/DependencyResolutionVerifierTest.groovy

```diff
@@ -751,6 +751,381 @@ class DependencyResolutionVerifierTest extends IntegrationTestKitSpec {
+                        "transitive": [
+                            "test.nebula:a"
+                        ]
+                    },
+                    "test.nebula:e": {
+                        "firstLevelTransitive": [
+                            "${moduleName}:sub2"
+                        ],
+                        "locked": "3.1.0"
+                    },
+                    "test.nebula:f": {
+                        "firstLevelTransitive": [
+                            "${moduleName}:sub2"
+                        ],
+                        "locked": "3.2.0"
+                    },
+                    "test.nebula:g": {
+                        "firstLevelTransitive": [
+                            "${moduleName}:sub2"
+                        ],
+                        "locked": "3.1.0"
+                    },
+                    "test.nebula:h": {
+                        "locked": "3.1.0",
+                        "transitive": [
+                            "test.nebula:e"
+                        ]
+                    }
+                }
+            }""".stripIndent()
+        new File(projectDir, 'global.lock').text = globalLockText
+
+        when:
+        def flags = ["-Dnebula.features.coreAlignmentSupport=${coreAlignment}", "-Dnebula.features.coreLockingSupport=${coreLocking}"]
+        def results = runTasksAndFail('dependenciesForAll', '--configuration', 'compileClasspath', *flags)
+
+        then:
+        results.output.contains('FAILED')
+
+        results.output.contains("Dependency lock state is out of date:")
+
+        results.output.contains("Resolved 'test.nebula:a:1.2.0' instead of locked version '1.1.0' for project 'sub1'")
+        results.output.contains("Resolved 'test.nebula:c:1.2.0' instead of locked version '1.1.0' for project 'sub1'")
+        results.output.contains("Resolved 'test.nebula:d:1.2.0' instead of locked version '1.1.0' for project 'sub1'")
+
+        results.output.contains('Please update your dependency locks or your build file constraints.')
+
+        when:
+        def updateGlobalLocks = runTasks('deleteGlobalLock', 'generateGlobalLock', 'saveGlobalLock', '-PdependencyLock.includeTransitives=true', *flags)
+        def updatedDependencies = runTasks('dependenciesForAll', '--configuration', 'compileClasspath', *flags)
+
+        then:
+        updatedDependencies.output.contains('SUCCESS')
+
+        updatedDependencies.output.contains('test.nebula:a:1.1.0 -> 1.2.0\n')
+        updatedDependencies.output.contains('test.nebula:b:1.2.0\n')
+        updatedDependencies.output.contains('test.nebula:c:1.1.0 -> 1.2.0\n')
+        updatedDependencies.output.contains('test.nebula:d:1.2.0\n')
+
+        where:
+        coreAlignment | coreLocking
+        true          | false
+    }
+
+    @Unroll
+    def 'resolved versions are equal to locked versions - with a substitution in place - core alignment #coreAlignment - core locking #coreLocking'() {
```

Comment:
This is the test case that kicked this off: when a substitution rule is in effect and the locked version uses the substitution DSL -> this should be fine and not fail due to multiple forces for an aligned platform.

In this case, Jackson is in the dependency graph, and there is a substitution rule for a micro-patch version. Alignment and locking should work without failing.